src/main/java/xyz/nucleoid/extras/lobby/contributor/ContributorData.java

```diff
@@ -0,0 +1,148 @@
+package xyz.nucleoid.extras.lobby.contributor;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.math.BigInteger;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.UUID;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.gson.JsonElement;
+import com.mojang.serialization.JsonOps;
+
+import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
+import net.minecraft.nbt.NbtCompound;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.util.JsonHelper;
+import xyz.nucleoid.extras.NucleoidExtrasConfig;
+import xyz.nucleoid.extras.lobby.block.ContributorStatueBlockEntity;
+import xyz.nucleoid.extras.mixin.lobby.ThreadedAnvilChunkStorageAccessor;
+
+public final class ContributorData {
+    private static final Logger LOGGER = LogManager.getLogger(ContributorData.class);
+
+    private static final Executor EXECUTOR = Executors.newSingleThreadExecutor(new ThreadFactoryBuilder()
+            .setNameFormat("contributor-data-fetcher")
+            .setDaemon(true)
+            .build()
+    );
+
+    private static final String PEOPLE_KEY = "people";
+    private static final String NAME_KEY = "name";
+    private static final String SOCIALS_KEY = "socials";
+    private static final String MINECRAFT_KEY = "minecraft";
+    private static final String STATUE_NBT_KEY = "statue_nbt";
+
+    private static final Map<String, Contributor> CONTRIBUTORS = new HashMap<>();
+
+    // Methods for accessing data
+
+    public static Iterable<Entry<String, Contributor>> getContributors() {
+        return CONTRIBUTORS.entrySet();
+    }
+
+    public static Contributor getContributor(String id) {
+        return CONTRIBUTORS.get(id);
+    }
+
+    // Methods for reloading data
+
+    private static void addContributor(Entry<String, JsonElement> entry) {
```

Comment:
Can this use Codec?

Reply:
Fixed in 9c01e272c7d6f353181ba1830dd3e02414f11429